From 47a3e104c56f4a3921abeee5aee629c8998898ad Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 08 Mar 2021 06:05:56 +0200
Subject: [PATCH] [SW-0] post-image scripts: fix compilation issue

Fix compilation issue after adding FIT_LOAD_TYPE_FLASH_READ.
grep started failing in scripts after adding additional load type,
as when searching for FIT_LOAD_TYPE_FLASH it gives multiple(2) matches
and sed fails.
Changed scripts to search for exact word.

Change-Id: I33ae5ff5b6e00b5e5a8b5d7ecc51f07c5127a738
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-agents.sh b/board/habanalabs/common/post-image-agents.sh
index d69556b..34ce3cb 100755
--- a/board/habanalabs/common/post-image-agents.sh
+++ b/board/habanalabs/common/post-image-agents.sh
@@ -25,7 +25,7 @@
 # Set hl-btl-version = <git data>; inside FIT header
 HL_BTL_VERSION=`cat ${BASE_DIR}/build/btl-custom/localversion-habanalabs`
 # Set enum number of FIT_LOAD_TYPE_FLASH in FIT
-FIT_LOAD_TYPE_FLASH=$(grep -E "LOAD_TYPE_FLASH[^_]" ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
+FIT_LOAD_TYPE_FLASH=$(grep -w FIT_LOAD_TYPE_FLASH ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 # Set compression to lzo
 HL_COMPRESS=lzo
 
diff --git a/board/habanalabs/common/post-image-fit-global.sh b/board/habanalabs/common/post-image-fit-global.sh
index c1d336c..8a27c15 100755
--- a/board/habanalabs/common/post-image-fit-global.sh
+++ b/board/habanalabs/common/post-image-fit-global.sh
@@ -29,7 +29,7 @@
 # Set hl-btl-version = <git data>; inside FIT header
 HL_BTL_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_NAME}/build/btl-custom/localversion-habanalabs`
 # Set enum number of FIT_LOAD_TYPE_FLASH in FIT
-FIT_LOAD_TYPE_FLASH=$(grep LOAD_TYPE_FLASH ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
+FIT_LOAD_TYPE_FLASH=$(grep -w FIT_LOAD_TYPE_FLASH ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 # Set the security key index and version
 HL_SEC_FILE=$SPECS_EMBEDDED_ROOT/${DEV_NAME}/${DEV_NAME}_security.h
 HL_SEC_KEY_INDEX=`grep HL_SEC_KEY_INDEX $HL_SEC_FILE | sed 's/^.*\(0x.*\)$/\1/'`
diff --git a/board/habanalabs/common/post-image-fit.sh b/board/habanalabs/common/post-image-fit.sh
index 2bf465b..e105a13 100755
--- a/board/habanalabs/common/post-image-fit.sh
+++ b/board/habanalabs/common/post-image-fit.sh
@@ -27,7 +27,7 @@
 # Set hl-btl-version = <git data>; inside FIT header
 HL_BTL_VERSION=`cat ${BASE_DIR}/build/btl-custom/localversion-habanalabs`
 # Set enum number of FIT_LOAD_TYPE_FLASH in FIT
-FIT_LOAD_TYPE_FLASH=$(grep -E "FIT_LOAD_TYPE_FLASH[^_]" ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
+FIT_LOAD_TYPE_FLASH=$(grep -w FIT_LOAD_TYPE_FLASH ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 # Set the security key index and version
 HL_SEC_FILE=$SPECS_EMBEDDED_ROOT/${HL_DEV}/${HL_DEV}_security.h
 HL_SEC_KEY_INDEX=`grep HL_SEC_KEY_INDEX $HL_SEC_FILE | sed 's/^.*\(0x.*\)$/\1/'`
